harness/determined/tensorboard/fetchers/azure.py

```diff
@@ -0,0 +1,76 @@
+import datetime
+import logging
+import os
+import urllib
+from typing import Any, Dict, Generator, List, Tuple
+
+from .base import Fetcher
+
+logger = logging.getLogger(__name__)
+
+
+class AzureFetcher(Fetcher):
+    def __init__(self, storage_config: Dict[str, Any], storage_paths: List[str], local_dir: str):
+        from azure.storage import blob
+
+        connection_string = storage_config.get("connection_string", None)
+        container = storage_config.get("container", None)
+        account_url = storage_config.get("account_url", None)
+        credential = storage_config.get("credential", None)
+
+        if storage_config.get("connection_string", None):
+            self.client = blob.BlobServiceClient.from_connection_string(connection_string)
+        elif account_url:
+            self.client = blob.BlobServiceClient(account_url, credential)
+        else:
+            raise ValueError("Either 'container_string' or 'account_url' must be specified.")
+
+        self.container_name = container if not container.endswith("/") else container[:-1]
+
+        self.local_dir = local_dir
+        self.storage_paths = storage_paths
+        self._file_records = {}  # type: Dict[str, datetime.datetime]
+
+    def _list(self, prefix: str) -> Generator[Tuple[str, datetime.datetime], None, None]:
+        container = self.client.get_container_client(self.container_name)
+        logger.debug(f"Listing keys in container '{self.container_name}' with '{prefix}'")
+
+        prefix = urllib.parse.urlparse(prefix).path.lstrip("/")
+
+        blobs = container.list_blobs(name_starts_with=prefix)
+        logger.debug(f"list_blobs response obj: {blobs}")
+        for blob in blobs:
+            yield (blob["name"], blob["last_modified"])
+
+    def fetch_new(self) -> int:
+        new_files = []
+
+        # Look at all files in our storage location.
+        for storage_path in self.storage_paths:
+            logger.debug(f"Looking at path: {storage_path}")
+
+            for filepath, mtime in self._list(storage_path):
+                prev_mtime = self._file_records.get(filepath, None)
+
+                if prev_mtime is not None and prev_mtime >= mtime:
+                    logger.debug(f"Blob not new: '{filepath}'")
```

Comment:
most of the `logger.debug()` statements in these `fetchers/*` files strike me as "useful while developing the feature but would probably just be noise if I turned logging to debug in real life".

You do more support than I do, so if you feel these should stay I won't press the issue.  But I want you to consider them carefully, because unhelpful logging is not without cost.  This one in particular seems like it would be in a pretty hot loop, but would be particularly useless.

Reply:
I agree. Removed.